llmfoundry/callbacks/curriculum_learning_callback.py

```diff
@@ -34,20 +40,188 @@ class CurriculumLearning(CallbackWithConfig):
         dataset_index (int): The index of the dataset currently being used.
     """
 
-    def __init__(self, train_config: Dict, dataset_index: int):
-        self.dataset_index = dataset_index
-        self.saved_dataset_index = 0
-        self.all_dataset_configs = []
-        self.current_dataset_state = {}
-        # The current dataset config is resolved and passed in train.py
-        self.current_dataset_config = train_config['train_loader']
+    def __init__(
+        self,
+        train_config: dict[str, Any],
+        schedule: list[dict[str, Any]],
+    ):
+        from llmfoundry.utils.builders import build_tokenizer
+        from llmfoundry.utils.config_utils import calculate_batch_size_info
+
+        # Ensure all duration values are positive
+        # Ensure all duration units are in epochs or tokens
+        self._schedule = schedule
+        if len(self._schedule) == 0:
+            raise ValueError('The schedule must have at least one datamix.')
+        for index, datamix in enumerate(self._schedule):
+            self._validate_data_mix(datamix)
```

Comment:
This fn is called `_validate_datamix` in the code

Reply:
Done